db/migrate/20210209172653_create_course_lesson_progresses.rb

```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+class CreateCourseLessonProgresses < ActiveRecord::Migration[6.1]
+  def change
+    create_table :course_lesson_progresses, id: :uuid do |t|
+      t.string :progress, default: "not_started"
```

Comment:
If this only has a fixed set of values then probably we should use an enum rather than a string field. https://api.rubyonrails.org/v5.2.4.4/classes/ActiveRecord/Enum.html

Reply:
I prefer to store enums as strings over integers, but that solution looks like it would work: https://stackoverflow.com/a/32939934

Reply:
+1 for string enums

Reply:
Preferably stored as an enum in the db

Reply:
Something like https://til.hashrocket.com/posts/b492fb26ec-rails-enums-and-postgresql-enums ?

I think this is more effort than it's worth. We will be changing this many times over (I think), and I don't want each change to deal with the postgres type. Also I don't like to rely on database-specific implementations, especially when they go over the head of the ORM we are using.

Plus, I would be shocked if Rails understands postgres types and outputs them into schema file. And I quite like being able to see the schema and know what is in the db.

Also, with an enum, trying to assign an incorrect value will raise a nice exception, so I think it's pretty safe and we will catch it early enough.